### PR TITLE
Implemented Maybe and Single merge

### PR DESCRIPTION
**Pull request by arkivanov**

### CODE REVIEW

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/Merge.kt [hunk 1]
```diff
@@ -1,11 +1,43 @@
...
+            activeSourceCount.addAndGet(1)
+            upstream.subscribe(
+                object : ObjectReference<Disposable?>(null), CompletableObserver, ErrorCallback by serializedEmitter {
+                    override fun onSubscribe(disposable: Disposable) {
+                        value = disposable
+                        disposables += disposable
+                    }
+
+                    override fun onComplete() {
+                        disposables -= this.value!!
```
Comment (CherryPerry) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/Merge.kt]: Maybe we can avoid `!!`?
`this.value?.also { disposables -= 1 }`

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/maybe/Merge.kt [hunk 2]
```diff
@@ -1,15 +1,53 @@
...
+                        disposables += disposable
+                    }
+
+                    override fun onSuccess(value: T) {
+                        serializedEmitter.onNext(value)
+                        onComplete()
+                    }
+
+                    override fun onComplete() {
+                        disposables -= this.value!!
```
Comment (CherryPerry) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/maybe/Merge.kt]: Same.

File: reaktive/src/commonTest/kotlin/com/badoo/reaktive/single/Merge.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                object : ObjectReference<Disposable?>(null), SingleObserver<T>, ErrorCallback by serializedEmitter {
+                    override fun onSubscribe(disposable: Disposable) {
+                        value = disposable
+                        disposables += disposable
+                    }
+
+                    override fun onSuccess(value: T) {
+                        serializedEmitter.onNext(value)
+
+                        disposables -= this.value!!
```
Comment (CherryPerry) [inline reaktive/src/commonTest/kotlin/com/badoo/reaktive/single/Merge.kt]: Same.

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/Merge.kt (hunk 1 above)
Comment (arkivanov) [reply reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/Merge.kt]: I will use requreNotNull instead

**Pull request merged**